tools/fmt/fmt.rs

```diff
@@ -137,13 +145,18 @@ fn fold(
     }
 }
 
+enum SyntaxMatch {
+    NotFound,
+    Found(SyntaxKind),
+}
+
 fn whitespace_to(
     sub: &mut impl Iterator<Item = NodeOrToken>,
     element: SyntaxKind,
     writer: &mut impl TokenWriter,
     state: &mut FormatState,
     prefix_whitespace: &str,
-) -> Result<bool, std::io::Error> {
+) -> Result<SyntaxMatch, std::io::Error> {
```

Comment:
I believe here that the `Found` would always be the `element` , so it could still return a bool and avoiding matches in the caller.